### PR TITLE
Add support for python up to 3.11

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*

*Description of changes:*
- Add support for python up to 3.11

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/pr-ci.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python: [ 3.7, 3.8, 3.9 ]
+        python: [ 3.8, 3.9, "3.10", "3.11" ]
```
Comment (mircealam) [inline .github/workflows/pr-ci.yml:16]: is there a reason 3.10 and 3.11 are in quotes?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -28,7 +28,7 @@ repos:
   - id: check-merge-conflict
   - id: check-yaml
 - repo: https://github.com/pycqa/flake8
-  rev: "5.0.4"
```
Comment (mircealam) [inline .pre-commit-config.yaml:31]: same observation as above? (quote in all the places or quotes in no places?)

File: python/rpdk/python/codegen.py [hunk 3]
```diff
@@ -425,19 +424,25 @@ def _pip_build(cls, base_path):
...
+class Python310LanguagePlugin(_PythonLanguagePlugin):
+    NAME = "python310"
+    RUNTIME = "python3.10"
+    DOCKER_TAG = 3.10
+
+
+class Python311LanguagePlugin(_PythonLanguagePlugin):
+    NAME = "python311"
+    RUNTIME = "python3.11"
+    DOCKER_TAG = 3.10
```
Comment (mircealam) [inline python/rpdk/python/codegen.py:448]: is 3.10 the right tag?

File: .github/workflows/pr-ci.yml (hunk 1 above)
Comment (kddejong) [reply .github/workflows/pr-ci.yml:16]: I will switch them all to double quotes.  For some reason when we started adding two digits behind the `.` we needed to add quotes.

File: python/rpdk/python/codegen.py (hunk 3 above)
Comment (kddejong) [reply python/rpdk/python/codegen.py]: Fixed

File: .pre-commit-config.yaml (hunk 2 above)
Comment (kddejong) [reply .pre-commit-config.yaml:31]: removed quotes

File: .github/workflows/pr-ci.yml (hunk 1 above)
Comment (kddejong) [reply .github/workflows/pr-ci.yml]: Added in double quotes to all values since it is required in 3.10

**Review by mircealam — APPROVED** [review]

**Pull request merged**